<launch>

  <!-- defines name of the namespace of the drone -->
  <arg name="UAV_NAME" default="$(env UAV_NAME)" />
  <arg name="RUN_TYPE" default="$(env RUN_TYPE)" />

  <!-- set to false if the nodelet should run under an external nodelet manager -->
  <arg name="standalone" default="true" />

  <!-- allows disabling the gui from command line -->
  <arg name="gui" default="true" />

  <!-- will it run using GNU debugger? -->
  <arg name="debug" default="false" />
  <arg unless="$(arg debug)" name="launch_prefix_debug" value=""/>
  <arg     if="$(arg debug)" name="launch_prefix_debug" value="debug_roslaunch"/>

  <!-- will it run as standalone nodelet or using a nodelet manager? -->
  <arg name="manager" default="$(arg UAV_NAME)_basler_stereopair_driver_manager" />
  <arg     if="$(arg standalone)" name="nodelet" value="standalone"/>
  <arg unless="$(arg standalone)" name="nodelet" value="load"/>
  <arg     if="$(arg standalone)" name="nodelet_manager" value=""/>
  <arg unless="$(arg standalone)" name="nodelet_manager" value="$(arg manager)"/>

  <!--  -->
  <arg name="stereopair_base_frame_id" default="$(arg UAV_NAME)/basler_stereopair/base" />
  <arg name="camera1_fleft_pose" default="/$(arg UAV_NAME)/basler_stereopair/camera1_fleft_pose" />
  <arg name="camera2_fright_pose" default="/$(arg UAV_NAME)/basler_stereopair/camera2_fright_pose" />

  <!-- Namespace - all topics and parameters will be remapped using this namespace as a prefix (eg.: "/vision_example/image_in" to "/uav1/vision_example/image_in") -->
  <group ns="$(arg UAV_NAME)">

    <!-- ClASS_NAME nodelet -->
    <node pkg="nodelet" type="nodelet" name="basler_stereo_driver" args="$(arg nodelet) basler_stereo_driver/BaslerStereoDriver $(arg nodelet_manager)" launch-prefix="$(arg launch_prefix_debug)" output="screen">

      <!-- Parameters loaded from launch file -->
      <param name="gui" type="bool" value="$(arg gui)"/>

      <param name="UAV_NAME" type="string" value="$(arg UAV_NAME)"/>
      <param name="fwd_left/rect_topic_name" type="string" value="/$(arg UAV_NAME)/basler_left/image_rect" />
      <param name="fwd_right/rect_topic_name" type="string" value="/$(arg UAV_NAME)/basler_right/image_rect" />

      <param name="camera1_fleft_pose" type="string" value="$(arg camera1_fleft_pose)" />
      <param name="camera2_fright_pose" type="string" value="$(arg camera2_fright_pose)" />


      <!-- ROS parameters config file -->
      <rosparam file="$(find basler_stereopair_driver)/config/cameras_roi.yaml" />

      <!-- Subscribers (note that the topic names will also be remapped using the namespace as a prefix) -->

      <!-- Publishers (note that the topic names will also be remapped using the namespace as a prefix) -->

    </node>

    <node pkg="tf" type="static_transform_publisher" name="camera1_fleft_pose" args="0.05864 0.05864 0 -0.653 0.271 -0.271 0.653 $(arg stereopair_base_frame_id) $(arg camera1_fleft_pose) 100" />
    <node pkg="tf" type="static_transform_publisher" name="camera2_fright_pose" args="0.05864 -0.05864 0 -0.271 0.653 -0.653 0.271 $(arg stereopair_base_frame_id) $(arg camera2_fright_pose) 100" />

  </group>

</launch>
